EIPS/eip-6046.md

```diff
@@ -0,0 +1,63 @@
+---
+eip: 6046
+title: Replace SELFDESTRUCT with DEACTIVATE
+description: Change SELFDESTRUCT to not delete storage keys and use a special value in the account nonce to signal deactivation
+author: Alex Beregszaszi (@axic)
+discussions-to: https://ethereum-magicians.org/t/almost-self-destructing-selfdestruct-deactivate/11886
+status: Draft
+type: Standards Track
+category: Core
+created: 2022-11-25
+requires: 2681
+---
+
+## Abstract
+
+Change `SELFDESTRUCT` to not delete all storage keys, and to use a special value in the account nonce to signal *decativated* accounts. Because the semantics of revival change (storage keys may exists), we also rename the instruction to `DEACTIVATE`.
+
+## Motivation
+
+The `SELFDESTRUCT` instruction currently has a fixed price, but is unbounded in terms of how many storage/account changes it performs (it needs to delete all keys). This is an outstanding concern for long.
+
+Furthermore with *Verkle trees* accounts will be organised differently: account properties, including storage, will have individual keys, and it will not be possible to traverse and find all used keys. This makes `SELFDESTRUCT` very challenging to support in Verkle trees.
+
+## Specification
+
+1. Change the rules introduced by [EIP-2681](./eip-2681.md) such that regular nonce increase is bound by `2^64-2` instead of `2^64-1`. This applies from genesis.
+
+2. The behaviour of `SELFDESTRUCT` is changed:
+
+  - Do not delete any storage keys and also leave the account in place.
```

Comment:
```suggestion
  - `SELFDESTRUCT` does not delete any storage keys 
  - `SELFDESTRUCT` does not change any account properties other than its balance
  - `SELFDESTRUCT` marks the account as warm
```

Reply:
I think it is a bit excessive to reiterate all prior rules (as they are complex), but instead left a note regarding EIP-2929.